firmware/globals.cpp

```diff
@@ -2,27 +2,45 @@
 #include "const.h"
 #include "globals.h"
 
+uint8_t absolute_mode = 1;
 uint8_t val;
 char buffer[BUFFER_SIZE];
 uint16_t buf_idx;
-
+uint16_t lineNumber = 0;
 // Current system state
-uint32_t position[2]    = {0,0};
-uint32_t destination[2] = {0,0}; // currently in steps
-uint16_t modal_feedrate = 0;
-int8_t direction[2] = {0,0};
-uint16_t debug_counter = 0;
+uint16_t buffleft=0;
+#ifndef Z_DRIVER_ENABLED
+uint32_t realposition[2] = {0, 0}; // distance in mm from absolute Zero
```

Comment:
Since you added the Ennn parameter, should there be a fourth axis for extruder position?